mlflow/spark.py

```diff
@@ -39,6 +41,47 @@ def log_model(spark_model, artifact_path, conda_env=None, jars=None):
                      jars=jars, conda_env=conda_env)
 
 
+def _tmp_path(len):
+    return ''.join(random.choice(string.ascii_uppercase + string.digits) for _ in range(len))
+
+
+class _DFS:
+    _fs = None
+
+    @classmethod
+    def jvm(cls):
+        return SparkContext._gateway.jvm
+
+    @classmethod
+    def fs(cls):
+        if not cls._fs:
+            sc = SparkContext.getOrCreate()
+            cls._fs = cls.jvm().org.apache.hadoop.fs.FileSystem.get(sc._jsc.hadoopConfiguration())
+        return cls._fs
+
+    @classmethod
+    def local_path(cls, path):
```

Comment:
Maybe we can differentiate the "public" API from the "private" API of this class, by using underscores. This also helps understand the contract. For example, I think only `copy_to_local`,  `copy_from_local`, and `remove` are exposed.

Maybe we could also name them more similarly to the Hadoop equivalents to make it more obvious; for example, `copy_to_local_file`, `copy_from_local_file`, and `delete`.